core/src/main/java/org/elasticsearch/index/query/SpanMultiTermQueryBuilder.java

```diff
@@ -62,11 +63,21 @@ protected void doXContent(XContentBuilder builder, Params params)
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
         Query subQuery = multiTermQueryBuilder.toQuery(context);
+        float boost = 1f;
+        if (subQuery instanceof BoostQuery) {
+            BoostQuery boostQuery = (BoostQuery) subQuery;
+            subQuery = boostQuery.getQuery();
+            boost = boostQuery.getBoost();
+        }
         if (subQuery instanceof MultiTermQuery == false) {
             throw new UnsupportedOperationException("unsupported inner query, should be " + MultiTermQuery.class.getName() +" but was "
                     + subQuery.getClass().getName());
         }
-        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) subQuery);
+        Query wrapper = new SpanMultiTermQueryWrapper<>((MultiTermQuery) subQuery);
+        if (boost != 1f) {
+            wrapper = new BoostQuery(wrapper, boost);
```

Comment:
good point...maybe... need to check why there are no failing tests around this, there should be. I think this is definitely a bug